### ISSUE TITLE
Create official instance

### ISSUE THREAD

**Issue by nfbyte** [Chocobozzz/PeerTube#5931]
### Introduction 

I think the PeerTube users and project itself would benefit significantly from having an official default instance with an associated official Patreon creator page to cover the infrastructure costs of running the instance as well as to bolster the project's development (much like Mastodon).

* * *

### Primary benefits

- Improve PeerTube's usability by enabling a sign-up process with a default instance
- Allow for testing PeerTube (in production), making the application more robust and helping identify scalability problems
- Showcase that the software works in practice, not just in theory (relevant for prospective instance admins)

* * *

### Funding

Crowdfunding (via e.g. Patreon / Open Collective), I believe, is the best way to finance both the instance and the project itself (without resorting to implementing anti-features such as advertising and content monetization) due to its scalability: the supporter base grows proportionally to the user base, which scales with the operational and development costs. Any potential overflow funds can be invested back into the open source projects relied on to make everything work (i.e. The Linux Foundation, The OpenJS Foundation, etc.).

As a strategy to manage operational costs, usage limits can be imposed on free users and lessened for certain tiers of support (as an example: free users get 1TB of storage for their videos, tier 1 supporters get 1.5TB, tier 2 get 3TB etc.), and other benefits can be trivially introduced for supporters such as a badge (#1583) next to their channel name.

* * *

### Addressing potential misconceptions

> Wouldn't this decrease decentralization?

No. PeerTube is decentralized by design through its open source development model (and its implementation of the ActivityPub protocol). Having an official instance would not change that. In fact, by making the software easier to use for more people and showcasing its capabilities, more instances would end up being created which if anything would *increase* decentralization.

> Wouldn't it be impossible to moderate?

No. I think it would make the most sense for a default instance to disallow NSFW content (which, along with GDPR compliance (#226), should solve the vast majority of potential non-copyright-related legislative issues) through a mix of automated and manual moderation, with the bulk of the work being automated (as detailed below).

* * *

### Machine Learning

An official instance would additionally serve as a practical opportunity to add support for running Machine Learning tasks through PeerTube's [remote runner](https://docs.joinpeertube.org/admin/remote-runners) architecture: for example, aside from transcoding using `ffmpeg`, PeerTube could dispatch additional post-processing tasks such as moderation (through [image classification](https://huggingface.co/tasks/image-classification)) and captioning / transcription (through [speech recognition](https://huggingface.co/tasks/automatic-speech-recognition) and [translation](https://huggingface.co/tasks/translation)).

Open source models for these tasks already exist and are available (e.g. on [Hugging Face](https://huggingface.co/models)); for NSFW detection specifically, an open source model based on [MobileNet V2](https://huggingface.co/docs/transformers/model_doc/mobilenet_v2) is available [here](https://huggingface.co/nfbyte/nsfw_mobilenet_v2_140_224), and for speech recognition, there is the [Whisper](https://huggingface.co/docs/transformers/model_doc/whisper) model. These models should theoretically be usable without the need to implement complex wrappers around Python commands via [transformers.js](https://github.com/xenova/transformers.js) ([example](https://huggingface.co/docs/transformers.js/tutorials/node-audio-processing)).
 
Video data could be split into image data (for moderation) and audio data (for captioning) and sent to the runners to be processed in parallel.

* * *

### First steps

- [ ] Extend PeerTube remote runner to support ML tasks
- [ ] Create a PeerTube task force with several divisions: Development, Infrastructure, Relations
- [ ] Create associated Matrix channels for each division (e.g. `#peertube-infrastructure`)
- [ ] Discuss and decide deployment strategy and details (Kubernetes vs. Hashistack, etc.)
- [ ] Decide on a domain name
- [ ] Spin up preliminary infrastructure and launch crowdfunding page
- [ ] Formalize and hire infrastructure and relations teams into Framasoft (or a separate legal entity)

* * *

I've also proposed this on Matrix [here](https://matrix.to/#/!LTiCLvjZggzbXFVQjA:matrix.tcit.fr/$2XTvDGU5E3af31--wi2GZF7Z43MpHbQp93quylHyEFQ?via=libera.chat&via=matrix.org&via=matrix.fdn.fr).

**Comment by EchedeyLR** [comment]
I want to give my two cents here and I would qualify that this may be focused on free platforms first (OpenCollective and LiberaPay) and maybe add the rest as additions by number of requests.

**Comment by nfbyte** [comment]
> It sounds like you have a great plan for setting up your own patreon funded instance, which could be a featured instance at joinpeertube.org if successful. Seems completely out of scope for Framasoft and the PeerTube team and a poor use of their limited resources and time

I don't personally *want* to set up an instance. I want to be able to use one, and there is no obvious choice, no entrypoint. I don't want to have to do research on which instance to choose (e.g. which one is trustworthy, which one is reliable (#5783), etc.) to get started, and I'm pretty sure most people don't either.

**Comment by EchedeyLR** [comment]
@nfbyte what you don't want to do is exactly what I feel that should be performed.

Search for your own or deploy one. That is what the Fediverse is for (and avoid using Mastodon Social as an example of such a thing because is neither and also a failure).

**Comment by ROBERT-MCDOWELL** [comment]
@nfbyte why not to make it yourself and share it to the community?

**Comment by Chocobozzz** [comment]
Hi,

Thanks for the discussion. You can see in the [v6 roadmap blog post](https://joinpeertube.org/news/roadmap-v6) that we plan to have an "official" peertube instance so we can highlight interesting peertube content. However this instance won't have registration opened because Framasoft don't have enough resources to moderate this kind of platform. I encourage to continue the discussion/proposal on https://ideas.joinpeertube.org/ instead :)

**Comment by nfbyte** [comment]
I feel like you have not even bothered to read what I wrote. No wonder that you don't have enough resources when you completely ignore proposals involving a [straight-forward plan to acquire resources](https://github.com/Chocobozzz/PeerTube/issues/5931#funding).

Continued in https://ideas.joinpeertube.org/posts/178/create-official-instance-with-registration-enabled.
